driver/src/it/scala/org/mongodb/scala/MongoCollectionCaseClassSpec.scala

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2016 MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.mongodb.scala
+
+import org.mongodb.scala.bson.codecs.Macros.createCodecProvider
+import org.bson.codecs.configuration.CodecRegistries.{fromRegistries, fromProviders}
+
+class MongoCollectionCaseClassSpec  extends RequiresMongoDBISpec {
+
+  case class Contact(phone: String)
+  case class User(_id: Int, username: String, age: Int, hobbies: List[String], contacts: List[Contact])
+  case class Optional(_id: Int, optional: Option[Int])
+
+  val codecRegistry = fromRegistries(fromProviders(classOf[User], classOf[Contact], classOf[Optional]), MongoClient.DEFAULT_CODEC_REGISTRY)
+
+  "The Scala driver" should "handle case classes" in withDatabase(databaseName) {
+    database =>
+      val collection = database.withCodecRegistry(codecRegistry).getCollection[User](collectionName)
```

Comment:
Its 6 of one and half a dozen of the other.  Its not a test of `withCodecRegistry` as I'd expect users to normally configure the registry when they create the MongoClient.